src/simsopt/geo/boozersurface.py

```diff
@@ -104,20 +105,29 @@ def __init__(self, biotsavart, surface, label, targetlabel, constraint_weight=No
                 - `bfgs_maxiter` (int): maximum number of iterations for BFGS solver. Defaults to 1500.
                 - `limited_memory` (bool): True if L-BFGS solver is desired, False if the BFGS solver otherwise. Defaults to False.
                 - `weight_inv_modB` (float): for BoozerLS surfaces, weight the residual by modB so that it does not scale with coil currents.  Defaults to True.
+                - solver (int): list of flags to decide what algorithm(s) to use when running BoozerSurface.run_code. Only used if boozer_type is 'ls'. Default is ['ls', 'ls_newton']
+                    bfgs: LBFGS applied to the penalty objective
+                    ls: scipy.optimize.least_squares applied to the penalty objective. Specify in options what method -  can be 'manual' or anything specified in scipy LS documentation.
+                    ls_newton: Newton applied to the penalty objective -  see _minimize_boozer_penalty_constraints_newton
+                    exact_newton: Newton applied to the constrained optimization problem -  see _minimize_boozer_exact_constraints_newton
         """
         super().__init__(depends_on=[biotsavart])
 
         from simsopt.geo import SurfaceXYZFourier, SurfaceXYZTensorFourier
         if not isinstance(surface, SurfaceXYZTensorFourier) and not isinstance(surface, SurfaceXYZFourier):
             raise Exception("The input surface must be a SurfaceXYZTensorFourier or SurfaceXYZFourier.")
 
+        if boozer_type not in ['ls', 'exact']:
+            raise ValueError("Invalid boozer_type - please choose between 'ls' and 'exact'")
+
         self.biotsavart = biotsavart
```

Comment:
can you add an exception that is raised if the surface does not have the correct quadrature points required by the BoozerExact algorithm when 'exact' is chosen?

Reply:
Not sure about this one. Where are the different cases described? Do we always requires 2mpol+1 and 2ttor+1 points, or is it different for non-stellarator symmetric cases?